### PR TITLE
Adds support for the barcode input field

### PR DESCRIPTION
**Pull request by dotMorten**
Allows rendering the new BarcodeScannerInputField. This is just the text field, but with a QR code icon button users can click.
At this point of implementation, no default action happens clicking this, but users can react to the BarcodeButtonClicked event on the FeatureForm and use for instance ZXing to launch a barcode scanner.
```cs
    private async void FeatureFormView_BarcodeButtonClicked(object sender, BarcodeButtonClickedEventArgs e)
    {
        try {
            e.FormElement.UpdateValue(await GetBarcodeScan());
        } catch {}
    }
```

### CODE REVIEW

**Review by prathameshnarkhede — APPROVED** [review]

File: src/Toolkit/Toolkit/UI/Controls/FeatureForm/FeatureFormView.cs [hunk 1]
```diff
@@ -357,6 +358,28 @@ internal bool OnFormAttachmentClicked(FormAttachment attachment)
...
+        /// <summary>
+        /// Raised when the Barcode Icon in a barcode element is clicked.
+        /// </summary>
+        /// <remarks>
+        /// <para>Set the <see cref="BarcodeButtonClickedEventArgs.Handled"/> property to <c>true</c> to prevent
+        /// any default code and perform your own logic.</para>
+        /// </remarks>
+        public event EventHandler<BarcodeButtonClickedEventArgs>? BarcodeButtonClicked;
+
+        internal bool OnBarcodeButtonClicked(FieldFormElement element)
```
Comment (mstefarov) [inline src/Toolkit/Toolkit/UI/Controls/FeatureForm/FeatureFormView.cs:371]: Should this button be clickable (or even visible) when the field is read-only?

File: src/Toolkit/Toolkit/UI/Controls/FeatureForm/TextFormInputView.Maui.cs [hunk 2]
```diff
@@ -63,13 +66,18 @@ private static object BuildDefaultTemplate()
             Border errorBorder = new Border() { StrokeThickness = 1, Stroke = new SolidColorBrush(Colors.Red), IsVisible = false };
             Grid.SetColumnSpan(errorBorder, 2);
             root.Add(errorBorder);
+            ImageButton barcodeButton = new ImageButton() { IsVisible = false, BorderWidth = 0, Source = new FontImageSource() { FontFamily = "calcite-ui-icons-24", Glyph = "\uE22F", Color = Colors.Black } };
```
Comment (mstefarov) [inline src/Toolkit/Toolkit/UI/Controls/FeatureForm/TextFormInputView.Maui.cs:69]: The icon is rather misaligned.  And in dark mode, it's almost invisible (black-on-dark-gray):

File: src/Toolkit/Toolkit/UI/Controls/FeatureForm/FeatureFormView.cs (hunk 1 above)
Comment (dotMorten) [reply src/Toolkit/Toolkit/UI/Controls/FeatureForm/FeatureFormView.cs:371]: Nope and it isn't: 

https://github.com/Esri/arcgis-maps-sdk-dotnet-toolkit/blob/c46755d8ab57f4dff03512e0b12ed6ce11da1fa0/src/Toolkit/Toolkit/UI/Controls/FeatureForm/TextFormInputView.cs#L95
https://github.com/Esri/arcgis-maps-sdk-dotnet-toolkit/blob/c46755d8ab57f4dff03512e0b12ed6ce11da1fa0/src/Toolkit/Toolkit/UI/Controls/FeatureForm/TextFormInputView.Maui.cs#L71
https://github.com/Esri/arcgis-maps-sdk-dotnet-toolkit/blob/c46755d8ab57f4dff03512e0b12ed6ce11da1fa0/src/Toolkit/Toolkit.WPF/UI/Controls/FeatureForm/FeatureFormView.Theme.xaml#L178-L180

**Review by mstefarov — APPROVED** [review]
Things are looking good. Now with some bonus bugfixes for TextFormInputView!

**Pull request merged**